### PR TITLE
drop mcm

### PR DESCRIPTION
**Pull request by kevin-lacoo**
**How to categorize this PR?**

/area open-source
/kind cleanup
/platform alicloud

**What this PR does / why we need it**:
The MachineControllerManagerDeployment feature gate in gardenlet has been promoted to GA in v1.83, so we can safely cleanup all code related to MCM management in the extensions.
**Which issue(s) this PR fixes**:
Follow-up of https://github.com/gardener/gardener-extension-provider-alicloud/pull/617
Related to https://github.com/gardener/gardener/issues/7594

**Special notes for your reviewer**:

**Release note**:

```other operator
The code related to `machine-controller-manager` management has been cleaned up because `gardenlet` is responsible for it since `gardener/gardener@v1.83`.
```

### CODE REVIEW

**Review by rfranzke — APPROVED** [review]
/lgtm

File: pkg/controller/worker/machine_controller_manager.go [hunk 1]
```diff
@@ -6,33 +6,12 @@ package worker
...
-	return map[string]interface{}{
-		"providerName": alicloud.Name,
-		"namespace": map[string]interface{}{
-			"uid": namespace.UID,
-		},
-		"podLabels": map[string]interface{}{
-			v1beta1constants.LabelPodMaintenanceRestart: "true",
-		},
-	}, nil
+func (w *workerDelegate) GetMachineControllerManagerChartValues(_ context.Context) (map[string]interface{}, error) {
```
Comment (rfranzke) [inline pkg/controller/worker/machine_controller_manager.go:11]: Second look: This should not be needed anymore.

File: pkg/controller/worker/machine_controller_manager.go [hunk 2]
```diff
@@ -6,33 +6,12 @@ package worker
...
+	return nil, nil
 }
 
 func (w *workerDelegate) GetMachineControllerManagerShootChartValues(_ context.Context) (map[string]interface{}, error) {
```
Comment (rfranzke) [inline pkg/controller/worker/machine_controller_manager.go:15]: Second look: This should not be needed anymore.

File: pkg/controller/worker/machine_controller_manager.go (hunk 1 above)
Comment (kevin-lacoo) [reply pkg/controller/worker/machine_controller_manager.go]: I've delete this file.

File: pkg/controller/worker/machine_controller_manager.go (hunk 2 above)
Comment (kevin-lacoo) [reply pkg/controller/worker/machine_controller_manager.go]: I've delete this file.

**Review by shaoyongfeng — APPROVED** [review]

**Pull request merged**